pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -302,6 +302,12 @@ private BrokerResponseNative handleSQLRequest(long requestId, String query, Json
     }
     if ((offlineTableName == null) && (realtimeTableName == null)) {
       // No table matches the request
+      if (_tableCache.getTableConfig(TableNameBuilder.REALTIME.tableNameWithType(rawTableName)) == null
+          && _tableCache.getTableConfig(TableNameBuilder.OFFLINE.tableNameWithType(rawTableName)) == null) {
+        LOGGER.info("Table not found for request {}: {}", requestId, query);
+        requestStatistics.setErrorCode(QueryException.TABLE_DOES_NOT_EXIST_ERROR_CODE);
+        return new BrokerResponseNative(QueryException.TABLE_DOES_NOT_EXIST_ERROR);
```

Comment:
Suggest putting another constant response in `BrokerResponseNative` similar to `EMPTY_RESULT` and `NO_TABLE_RESULT` (we should probably rename this)